sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/descriptor/InstrumentDescriptor.java

```diff
@@ -18,7 +27,8 @@ public static InstrumentDescriptor create(
       String unit,
       InstrumentType type,
       InstrumentValueType valueType) {
-    return new AutoValue_InstrumentDescriptor(name, description, unit, type, valueType);
+    return new AutoValue_InstrumentDescriptor(
+        name, description, unit, type, valueType, SourceInfo.fromCurrentStack());
```

Comment:
Thinking ahead to when there is file based configuration of metrics and `CustomSourceInfo` is needed, `SourceInfo` is probably going to need to be an argument here so that the code that is interpreting the config file and setting up the metrics can set the appropriate `SourceInfo`.

Reply:
That won't be the case for InstrumentDescriptor.  Config-file will only affect `View` and `MetricDescriptor`.

You're right, but not in this location.